### PR TITLE
Feat/MET-5362 Create API and update database for record tiers

### PR DESCRIPTION
**Pull request by JoanaCMS**

### CODE REVIEW

File: src/test/resources/get_records_tiers_response_body.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+[{"record-id":"/1/URN_NBN_SI_doc_B1HM2TA6","content-tier":"4","content-tier-before-license-correction":"4","license":"OPEN","metadata-tier":"A","metadata-tier-language":"C","metadata-tier-enabling-elements":"C","metadata-tier-contextual-classes":"A"},{"record-id":"/1/URN_NBN_SI_doc_35SZSOCF","content-tier":"4","content-tier-before-license-correction":"4","license":"OPEN","metadata-tier":"A","metadata-tier-language":"C","metadata-tier-enabling-elements":"C","metadata-tier-contextual-classes":"A"}]
```
Comment (stzanakis) [inline src/test/resources/get_records_tiers_response_body.txt:1]: Group those files under a directory, for organization perhaps?

**Review by stzanakis — CHANGES_REQUESTED** [review]
Seems okay, added a few comments for perhaps some small improvements.

Can we do something about the code smells introduced?

Sonar analysis is complaining about test coverage, is it something we can increase for this code, or is it because some pervious code did already not have coverage?

File: src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java [hunk 2]
```diff
@@ -177,7 +177,7 @@ private RecordInfo harvestOaiRecords(String datasetId, OaiHarvestData oaiHarvest
           oaiHarvestData.getMetadataformat());
       OaiRecord oaiRecord = oaiHarvester.harvestRecord(oaiRepository,
           oaiHarvestData.getOaiIdentifier());
-      RecordEntity recordEntity = new RecordEntity(null, oaiHarvestData.getOaiIdentifier(), datasetId, "", "");
+      RecordEntity recordEntity = new RecordEntity(null, oaiHarvestData.getOaiIdentifier(), datasetId, null, null, null, null, null, null, null);
```
[the remark this answers is not in the document]
Comment (JoanaCMS) [reply src/main/java/eu/europeana/metis/sandbox/service/workflow/HarvestServiceImpl.java:180]: Done!

**Comment by JoanaCMS** [comment]
I've tried my best to improve the code smells, and when it comes to coverage, from what I see it is because of some previous code

File: src/test/java/eu/europeana/metis/sandbox/service/record/RecordServiceImplTest.java [hunk 3]
```diff
@@ -4,23 +4,27 @@
 import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.anyLong;
-import static org.mockito.Mockito.anyString;
-import static org.mockito.Mockito.reset;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.*;
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/service/record/RecordServiceImplTest.java:7]: just check this one has .*

File: src/test/java/eu/europeana/metis/sandbox/service/workflow/IndexingServiceImplTest.java [hunk 4]
```diff
@@ -3,17 +3,18 @@
...
 
 import eu.europeana.indexing.Indexer;
 import eu.europeana.indexing.IndexingProperties;
 import eu.europeana.indexing.exception.IndexerRelatedIndexingException;
 import eu.europeana.indexing.exception.IndexingException;
 import eu.europeana.indexing.tiers.model.MediaTier;
 import eu.europeana.indexing.tiers.model.MetadataTier;
+import eu.europeana.indexing.tiers.model.TierClassifier;
 import eu.europeana.indexing.tiers.model.TierResults;
+import eu.europeana.indexing.tiers.view.*;
```
Comment (jeortizquan) [inline src/test/java/eu/europeana/metis/sandbox/service/workflow/IndexingServiceImplTest.java:16]: check also here the import it has `*`

**Review by jeortizquan — COMMENTED** [review]
There are some `.*` imports to check.

should we consider a script for updating record table? or we just recreate the table when this goes to production.
```
ALTER TABLE record
ADD COLUMN content_tier_before_license_correction  VARCHAR(3) NULL,
ADD COLUMN metadata_tier_language                  VARCHAR(3) NULL,
ADD COLUMN metadata_tier_enabling_elements         VARCHAR(3) NULL,
ADD COLUMN metadata_tier_contextual_classes        VARCHAR(3) NULL,
ADD COLUMN license                                 VARCHAR(13) NULL;
```

File: src/test/java/eu/europeana/metis/sandbox/service/record/RecordServiceImplTest.java (hunk 3 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/service/record/RecordServiceImplTest.java:7]: Fixed

File: src/test/java/eu/europeana/metis/sandbox/service/workflow/IndexingServiceImplTest.java (hunk 4 above)
Comment (JoanaCMS) [reply src/test/java/eu/europeana/metis/sandbox/service/workflow/IndexingServiceImplTest.java:16]: Fixed

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**